### PR TITLE
Fix classname typo in tests

### PR DESCRIPTION
**Pull request by szepeviktor**
From #2458

### CODE REVIEW

File: tests/PHPStan/Type/Generic/GenericObjectTypeTest.php [hunk 1]
```diff
@@ -63,7 +63,7 @@ public function dataIsSuperTypeOf(): array
 				TrinaryLogic::createYes(),
 			],
 			'implementation with @extends with different type args' => [
-				new GenericObjectType(B\I::class, [new ObjectType('DateTimeInteface')]),
+				new GenericObjectType(B\I::class, [new ObjectType('DateTimeInterface')]),
```
Comment (staabm) [inline tests/PHPStan/Type/Generic/GenericObjectTypeTest.php:66]: Would `DateTimeInterface::class` work?

Comment (szepeviktor) [reply tests/PHPStan/Type/Generic/GenericObjectTypeTest.php:66]: I think yes.
But I only do what I'm told.

**Comment by ondrejmirtes** [comment]
Thank you.

**Comment by szepeviktor** [comment]
You're welcome.

Advancing with the next one 🏃🏻‍♂️

**Pull request merged**